### ISSUE TITLE
Configuration conflict with microbundle

### ISSUE THREAD

**Issue by briancavalier** [TylorS/typed-test#9]
I just tried `@typed/test` in [`@most/index`](https://github.com/mostjs/index/pull/3), and it worked great.  I ran into a snag, though.

Currently, `@typed/test` requires tsconfig to have module `"commonjs"`, and [microbundle](https://github.com/developit/microbundle) requires it to have module `"es2015"` or `"esnext"`.  As far as I can tell, neither allows specifying an alternate TS config, so I haven't found a way to use the two in the same project.

Here's an example of a Travis run where `microbundle` failed: https://travis-ci.org/mostjs/index/jobs/514103007.

Any ideas or workarounds for how to use both?

**Comment by briancavalier** [comment]
Maybe I can use `prepublishOnly` instead of `prepare` to avoid the `npm build` in CI.  I'll try that and see what happens.

**Comment by briancavalier** [comment]
Using `prepublishOnly` [does avoid the problem in CI](https://travis-ci.org/mostjs/index/jobs/514116945).  However, I think it only postpones the issue: `microbundle` will still fail, so as is, the package can't be published, since `npm publish` will try to run `prepublishOnly`.

**Comment by Frikki** [comment]
A current workaround is to have a base tsconfig file that can be extended. If the module is set in the base tsconfig, then there won’t be a conflict. For example:

**tsconfig.base.json**

```json
{
  "compilerOptions": {
    "target": "es5",
    "module": "es2015",
    "moduleResolution": "node",
    "lib": ["es2015", "es2016", "es2017"],
    "declaration": true,
    "sourceMap": true,
    "strict": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noImplicitAny": true,
    "noImplicitReturns": true,
    "noFallthroughCasesInSwitch": true,
    "strictNullChecks": true,
    "strictFunctionTypes": true,
    "allowSyntheticDefaultImports": true,
    "esModuleInterop": true
  },
  "exclude": ["node_modules", "src/**/*.__test__.ts"]
}
```

**tsconfig.json**

```json{
  "extends": "./tsconfig.base.json",
  "compilerOptions": {
    "outDir": "dist"
  },
  "include": ["src/**/*.ts"]
}
```

To avoid distributing test files, in the _package.json_:

```json
"build": "microbundle -i src/index.ts -f es,cjs"
```

**Comment by TylorS** [comment]
Hey @briancavalier I've published v9.1.0 that should fix this issue for you. Please give it a try when you can and let me know if all is well for you!

**Comment by briancavalier** [comment]
Thanks @TylorS.  Both the build and the tests work now 🎉 

<img width="505" alt="Screen Shot 2019-04-02 at 11 43 15 AM" src="https://user-images.githubusercontent.com/90518/55416392-b683cb00-553c-11e9-9e61-a1053ddc9d98.png">
